### PR TITLE
Tommy doing some workspace setup

### PR DESCRIPTION
**Pull request by TommyJones**

### CODE REVIEW

**Comment by TommyJones** [comment]
This code should run, though I've not tested it end-to-end yet. Planning to get re-acquainted with Google Cloud Compute first. Then I'll add scripts for that.

**Comment by TommyJones** [comment]
There are 5 R files and a folder necessary to run this (sequential and single metric) version. They are:
* data_derived/ - folder for storing saved objects (such as curated data and models)
* 00_globals.R - script that contains global variables and function
* 01_download_format_data.R - Downloads the 20 newsgroups data (if not already downloaded) and formats it into a document term matrix (which includes stop word removal etc.) and creates train/test/validation splits
*  02_lsa_sigopt.R - runs the SigOpt optimization loop for latent semantic analysis-based text going into a random forest classifier
* 03_lda_sigopt.R - runs the SigOpt optimization loop for latent dirichlet allocation-based text going into a random forest classifier
* 04_run_all.R - right now, despite its name, this script runs 02_lsa_sigopt.R and 03_lda_sigopt.R. 
* 05_run_cloudml.R - ships off the repo and data to Google Cloud Compute and runs 04_run_all.R

To run everything locally: Run in order: 01_download_format_data.R, 02_lsa_sigopt.R, and 03_lda_sigopt.R

As I write this, I realize that I'm going to have to rename/restructure the order of some of these files. Once 02_lsa_sigopt.R and 03_lda_sigopt.R are up and running, I'll have to build final models and have analysis in the 04_ + labeled scripts. At that point, I'll probably move the current 04_run_all.R and 05_run_cloudml.R outside of the numbering scheme. (e.g. I'll start their names with "_")

**Comment by TommyJones** [comment]
Code is done. Working on the text of the blog post now. Would love some feedback

File: 00_globals.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+### load packages ----
+library(textmineR)
+library(randomForest)
+library(magrittr)
+library(stringr)
+library(SigOptR)
+
+### declare GLOBAL variables ----
+random_seed <- 8675309
```
Comment (meghanaravikumar) [inline 00_globals.R]: Could you leave a comment for why this seed was picked?

Comment (meghanaravikumar) [inline 00_globals.R:13]: Could you leave a comment as to why this random seed was used?

File: 00_globals.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+predict_classifier <- function(object, new_data){
+  if (class(new_data) == "matrix")
+    new_data <- as.data.frame(new_data)
+  
+  predict(object = object, newdata = new_data, type = "prob")
+}
+
+# Function to get AUC from the random forest model
```
Comment (meghanaravikumar) [inline 00_globals.R:32]: does this have to be filled out?

File: 01_download_format_data.R [hunk 3]
```diff
@@ -0,0 +1,69 @@
+################################################################################
+# This script splits data into two training sets and a test set
+################################################################################
+
+rm(list = ls())
```
Comment (meghanaravikumar) [inline 01_download_format_data.R:5]: leave comment as to why the environment is being cleared

File: 01_download_format_data.R [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+set.seed(90210)
```
Comment (meghanaravikumar) [inline 01_download_format_data.R:7]: different seed?

File: 01_download_format_data.R [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  stringr::str_conv("UTF-8")
+
+names(docs) <- docnames
+
+doc_class <- stringr::str_split(docnames, pattern = "/") %>%
+  sapply(function(x) x[3])
+
+doc_class <- factor(doc_class)
+
+dtm <- CreateDtm(doc_vec = docs) # stopwords English and SMART
```
Comment (meghanaravikumar) [inline 01_download_format_data.R:44]: comment on why these words were used as stopwords to remove in the document would be useful

File: 01_download_format_data.R [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+dim(dtm)
+
+# prune vocab so that only words appearing in greater than 5 docs are included
+dtm <- dtm[, colSums(dtm > 0) >= 5]
```
Comment (meghanaravikumar) [inline 01_download_format_data.R:49]: good call

File: 01_download_format_data.R [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+dim(dtm)
+
+### sample rows into three groups ----
+
+idx <- seq_len(nrow(dtm))
+
+train1 <- sample(idx, 6665)
```
Comment (meghanaravikumar) [inline 01_download_format_data.R:57]: what's 6665?

File: 03_lda_sigopt.R [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  metrics <- list(list(name = "acc", value = acc), 
+                  list(name = "coh", value = coh))
+  
+  metrics
+  
+  
+}
+
+### run the optimization loop ----
+Sys.sleep(60)
```
Comment (meghanaravikumar) [inline 03_lda_sigopt.R:94]: comment as to why this sleep is necessary

File: 04_lda_optimal.R [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+### read in data ----
+load("data_derived/20_newsgroups_formatted.RData")
+
+### Get optimal parameters ----
+
+# hard coding this for now
+optimal_params <- list(k = 575,
+                       alpha = 0.01,
+                       beta_sum = 135.16)
+
```
Comment (meghanaravikumar) [inline 04_lda_optimal.R:23]: do you still want to leave this hardcoded?

**Review by meghanaravikumar — APPROVED** [review]

**Comment by meghanaravikumar** [comment]
Tommy could you write a ReadMe that walks through how to run the files?

**Pull request merged**